faculty/datasets/transfer.py

```diff
@@ -160,19 +155,57 @@ def upload_file(object_client, project_id, datasets_path, local_path):
     local_path : str
         The local path of the object to upload
     """
-    upload_stream(
+    file_size = os.path.getsize(local_path)
+    _upload_stream(
         object_client,
         project_id,
         datasets_path,
         _file_chunk_iterator(local_path),
+        known_file_size=file_size,
     )
 
 
-def _s3_upload(object_client, project_id, datasets_path, content, upload_id):
+def _upload_stream(
+    object_client, project_id, datasets_path, content, known_file_size=None
+):
 
-    completed_parts = []
+    presign_response = object_client.presign_upload(project_id, datasets_path)
 
-    for i, chunk in enumerate(_rechunk_data(content)):
+    if presign_response.provider == CloudStorageProvider.S3:
+        if known_file_size is None:
```

Comment:
There's a bit of duplication between here and below (both this if statement logic plus two other places where the `DEFAULT_CHUNK_SIZE` can be returned). How about merging all these cases together in a single `_chunk_size` helper like:

```python
def _chunk_size(provider, known_file_size):
    if provider == CloudStorageProvider.S3 and known_file_size is not None:
        return max([math.ceil(total_size / S3_MAX_CHUNKS), DEFAULT_CHUNK_SIZE])
    else:
        return DEFAULT_CHUNK_SIZE
```

Then, in this function, we can just call:

```
chunk_size = _chunk_size(presign_response.provider, known_file_size)
```

in one place before the if statement.